sklearn/utils/estimator_checks.py

```diff
@@ -413,11 +387,10 @@ def parametrize_with_checks(estimators):
     import pytest
 
     if any(isinstance(est, type) for est in estimators):
-        # TODO: remove class support in 0.24 and update docstrings
-        msg = ("Passing a class is deprecated since version 0.23 "
-               "and won't be supported in 0.24."
+        msg = ("Passing a class was deprecated in version 0.23 "
```

Comment:
I do not think we normally keep the deprecation warning version in the error message. Can this be:

> Starting from 0.23, only instances are supported.

Reply:
OK, I'll remove if others feel the same way. No strong opinion on my side, I just thought it might be useful since we only went for a 1 version deprecation cycle on this

Reply:
thoughts on this @rth?

Reply:
> Starting from 0.24, only instances are supported for the "estimators" parameter.

Yes +1 for this error message, mostly because it tells the user what to do when they get this error. It could be a TypeError instead of a ValueError maybe as well?